service/application/src/main/java/com/solace/maas/ep/event/management/agent/command/CommandManager.java

```diff
@@ -118,14 +122,22 @@ private static boolean exitEarlyOnFailedCommand(CommandBundle bundle, Command co
                 && (command.getResult() == null || JobStatus.error.equals(command.getResult().getStatus()));
     }
 
-    private void finalizeAndSendResponse(CommandMessage request) {
+    private void finalizeAndSendResponse(CommandRequest request) {
         request.determineStatus();
         Map<String, String> topicVars = Map.of(
-                "orgId", request.getOrgId(),
+                "orgId", eventPortalProperties.getOrganizationId(),
                 "runtimeAgentId", eventPortalProperties.getRuntimeAgentId(),
                 COMMAND_CORRELATION_ID, request.getCommandCorrelationId()
         );
-        commandPublisher.sendCommandResponse(request, topicVars);
+        CommandMessage response = new CommandMessage(request.getServiceId(),
+                request.getCommandCorrelationId(),
+                request.getContext(),
+                request.getStatus(),
+                request.getCommandBundles());
+        response.setOrgId(eventPortalProperties.getOrganizationId());
+        response.setTraceId(MDC.get(TRACE_ID));
+        response.setActorId(MDC.get(ACTOR_ID));
```

Comment:
We should try doing this in the mapper.
We'll probably have to add this to the no args constructor because that's what mapstruct will use by default (it will actually use whatever constructor has a `@Default` annotation if available - even if we make our own annotation).
```
withMessageType(MOPMessageType.generic)
                .withProtocol(MOPProtocol.epConfigPush)
                .withVersion("1")
                .withUhFlag(MOPUHFlag.ignore);
``` 

Then maybe have an after mapping for the traceId and actorId... and maybe even for the orgId...
```
@AfterMapping
    protected void assignMopMessageFields(@MappingTarget CommandMessage target, CommandRequest source) {
        target.setTraceId(MDC.get(TRACE_ID));
        // etc...
    }
```